packages/flutter/lib/src/animation/listener_helpers.dart

```diff
@@ -141,12 +141,14 @@ mixin AnimationLocalListenersMixin {
     for (final VoidCallback listener in localListeners) {
       InformationCollector? collector;
       assert(() {
-        collector = () sync* {
-          yield DiagnosticsProperty<AnimationLocalListenersMixin>(
-            'The $runtimeType notifying listeners was',
-            this,
-            style: DiagnosticsTreeStyle.errorProperty,
-          );
+        collector = () {
+          return <DiagnosticsNode>[
```

Comment:
might be better as `() => <DiagnosticsNode>[`

Reply:
(also applies other places)

Reply:
Style guide says that `=>` should only be used for single line methods. Even this one would be a pretty long line as a single line.

Reply:
Ah, I guess we except collection literals.. https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use--for-inline-callbacks-that-just-return-list-or-map-literals